File: Modules/hospital/Patients/PatientsColumns.cs
using Serene2.Modules.hospital.Patients.Enum;

namespace Serene2.hospital.Columns;

[ColumnsScript("hospital.Patients")]
[BasedOnRow(typeof(PatientsRow), CheckNames = true)]
public class PatientsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignCenter]
    public int PatientId { get; set; }

    [EditLink, Width(200)]
    public string PatientName { get; set; }

    [Width(100), AlignCenter]
    public int Age { get; set; }

    [Width(100), AlignCenter]
    public int Cost { get; set; }

    [Width(100), AlignCenter]
    public int TotalDiscount { get; set; }

    [Width(150)]
    public DateTime DateOfBirth { get; set; }

    [Width(100), AlignCenter]
    public GenderEnum Gender { get; set; }

    [Width(120)]
    public string Address { get; set; }

    [Width(120), QuickFilter]
    public string GovName { get; set; }

    [Width(120), QuickFilter]
    public string CityName { get; set; }
    public string DoctorName { get; set; }
}

File: Modules/ServerTypes/hospital/PatientsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GenderEnum } from "../Modules/hospital.Patients.Enum.GenderEnum";
import { PatientsRow } from "./PatientsRow";

export interface PatientsColumns {
    PatientId: Column<PatientsRow>;
    PatientName: Column<PatientsRow>;
    Age: Column<PatientsRow>;
    Cost: Column<PatientsRow>;
    TotalDiscount: Column<PatientsRow>;
    DateOfBirth: Column<PatientsRow>;
    Gender: Column<PatientsRow>;
    Address: Column<PatientsRow>;
    GovName: Column<PatientsRow>;
    CityName: Column<PatientsRow>;
    DoctorName: Column<PatientsRow>;
}

export class PatientsColumns extends ColumnsBase<PatientsRow> {
    static readonly columnsKey = 'hospital.Patients';
    static readonly Fields = fieldsProxy<PatientsColumns>();
}

[GenderEnum]; // referenced types